pwnlib/elf/elf.py

```diff
@@ -1200,6 +1200,22 @@ def search(self, needle, writable = False, executable = False):
             data   = self.mmap[offset:offset+memsz]
             data   += b'\x00' * zeroed
             offset = 0
+            if isinstance(needle, str) :
```

Comment:
Why not do `needle = _need_bytes(needle)` once outside the for-loop?

Reply:
oh , I will do that .